sdk/python/feast/data_source.py

```diff
@@ -596,6 +602,12 @@ def path(self):
         """
         return self._file_options.file_url
 
+    @property
+    def offline_store(self):
+        from feast.infra.offline_stores.file import FileOfflineStore
+
+        return FileOfflineStore()
```

Comment:
I am not sure if the approach being followed here is the correct one. Here we are saying "A source knows which store serves it", but wouldn't it be the reverse, like "A store knows which sources it can serve". So the way I was expecting the implementation to move towards was that the provider is given a list of sources and it decides based on the stores available to it which store to use.

Put differently, how would you allow two different store types to handle FileSource? Presumably we could have a Pandas implementation and a Dask implementation.